test/e2e/run_device_test.go

```diff
@@ -116,7 +126,27 @@ var _ = Describe("Podman run device", func() {
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(ExitCleanly())
 
-		createContainersConfFileWithDevices(podmanTest, "\"vendor.com/device=myKmsg\"")
+		createContainersConfFileWithDevices(podmanTest, "\"vendor.com/device=myKmsg\"", []string{})
+		session = podmanTest.Podman([]string{"run", "-q", "--security-opt", "label=disable", ALPINE, "test", "-c", "/dev/kmsg1"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+	})
+
+	It("podman run CDI device test with --cdi-spec-dir", func() {
+		cdiDir := podmanTest.TempDir + "/cdi"
+		if _, err := os.Stat(cdiDir); errors.Is(err, fs.ErrNotExist) {
+			Expect(os.MkdirAll(cdiDir, os.ModePerm)).To(Succeed())
+		}
```

Comment:
There is no need for the stat here, you can just MkdirAll() all the time (it will not error if the dir exist), also in this case we know the dir will never exist as this is our own private tempdir.